Micromouse/Maze.cpp

```diff
@@ -56,6 +56,14 @@ bool Maze::isWall(CellCoordinate pos, Direction dir) const {
     return isWall(pos.toNode() + dir);
 }
 
+bool Maze::isWall(NodeCoordinate pos, Direction dir) const {
```

Comment:
My original motivation for adding those functions is because I had a bug where I accidentally had `isWall(DirOp::relToDir(LEFT,facing))` instead of `isWall(position + DirOp::relToDir(LEFT,facing))`  and it compiled just fine. So I thought making the parameters separate would make it less likely to happen. Looking closer at the problem it is actually the `NodeCoordinate(int x = 0, int y = 0)` that is real issue. Direction is casted to int and then casted to NodeCoordinate with x=direction and y=0. This is definitely not what we want ever. Fixed in 11086ca